### PR TITLE
Fix bdb VLSN Inconsistency bug

### LINKED ISSUE #4969: FE restart failed with exception: recoveryTracker should overlap or follow on disk last VLSN of xxx recoveryFirst=xxx
### Steps to reproduce the behavior (Required)
ref: https://github.com/StarRocks/bdb-je/issues/1

### StarRocks version (Required)
 - all

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4969

## Problem Summary(Required) ：

We have fix the bdb VLSN Inconsistency bug in the 7.3.8. Update the maven dependency to that version. The problem is describe in https://github.com/StarRocks/bdb-je/issues/1
Since shutdown hook is a temporary solution to solve this bug, and it can be removed.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java [hunk 1]
```diff
@@ -137,45 +132,6 @@ public static void start(String starRocksDir, String pidDir, String[] args) {
         }
     }
 
-    // NOTE: To avoid dead lock
-    //      1. never call System.exit in shutdownHook
-    //      2. shutdownHook cannot have lock conflict with the function calling System.exit
-    private static void addShutdownHook() {
-        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:144]: Must add a more concise commit message about the problem and solution.
Don't make me guess what's the problem is.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java [hunk 2]
```diff
@@ -137,45 +132,6 @@ public static void start(String starRocksDir, String pidDir, String[] args) {
...
-            LOG.info("start to execute shutdown hook");
-            try {
-                Thread t = new Thread(() -> {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:147]: Why is the addShutdownHook() useless?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:144]: The problem is described in issue.

File: fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java (hunk 2 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/StarRocksFE.java:147]: Shutdown hook is a temporary solution to solve this bug. We have solve it in the bdb code, so it is not needed.

**Review by chaoyli — APPROVED** [review]

**Pull request merged**